File: tools/integrity/tests/fixtures/bad_contracts/common/common-py/gpusims_common/particle.py
"""ParticleFrame: deliberately mirrors the real ParticleFrame defect class."""

from dataclasses import dataclass, field


@dataclass
class ParticleFrame:
    positions: list[float] = field(default_factory=list)
    velocities: list[float] = field(default_factory=list)
    radii: list[float] = field(default_factory=list)
    ids: list[int] = field(default_factory=list)

File: tools/integrity/tests/fixtures/bad_contracts_b/common/common-web/src/particle.ts
// Mirrors ParticleFrame::radii defect: radii is declared but never read.
export class ParticleFrame {
    positions: number[] = [];
    velocities: number[] = [];
    radii: number[] = [];
    ids: number[] = [];

    constructor(n: number) {
        this.positions = new Array(n * 3);
        this.velocities = new Array(n * 3);
        this.radii = new Array(n);
        this.ids = new Array(n);
    }
}
